pandas/_libs/internals.pyx

```diff
@@ -16,6 +16,7 @@ cnp.import_array()
 from pandas._libs.algos import ensure_int64
 
 
+@cython.final
```

Comment:
do these actually make a diff?

Reply:
i havent measured this independently, so not sure how much it matters in this case (tried it for some Timestamp methods and got a decent boost).  it allows cython to do some inlining, at the cost of disallowing subclassing.